app/Domains/RevenueStreams/Actions/CreateRevenueStream.php

```diff
@@ -29,14 +31,23 @@ public function handle(array $params): RevenueStream
         return RevenueStream::create($params);
     }
 
-    public function rules(): array
+    public function rules(ActionRequest $request): array
     {
-        return [
+        $rules = [
             'name' => ['required', 'string', 'max:255'],
             'description' => ['nullable', 'string'],
             'type_id' => ['required', 'exists:revenue_stream_types,id'],
-            'values' => ['required', 'json'],
+            'values' => ['required', 'array'],
         ];
+
+        $type = RevenueStreamType::find($request->type_id);
+        if ($type) {
```

Comment:
Move this logic to withValidator, check [documentation](https://www.laravelactions.com/2.x/add-validation-to-controllers.html#custom-validation-logic) of withValidator.